papermill/hdfs.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+"""Utilities for working with S3."""
+import logging
+
+from pywebhdfs.errors import FileNotFound
+from pywebhdfs.webhdfs import PyWebHdfsClient
+
+from .s3 import retry
+
+logger = logging.getLogger('papermill.hdfs')
+
+# python 2 compatibility imports
+
+try:
+    FileNotFoundError
+except NameError:
+    FileNotFoundError = IOError
+
+
+class HDFS(object):
```

Comment:
We probably don't need to make all the HDFS operators prefixed with `_`. I'd say most of these are fine to expose without the private prefix.

Reply:
removed the private preix